File: Learner/src/main/java/edu/example/learner/courseabout/course/service/CourseService.java
package edu.example.learner.courseabout.course.service;

import edu.example.learner.courseabout.course.dto.CourseDTO;
import edu.example.learner.courseabout.course.dto.MemberCourseDTO;
import edu.example.learner.courseabout.course.entity.CourseAttribute;

import java.util.List;

public interface CourseService {
    CourseDTO addCourse(CourseDTO courseDTO);
    CourseDTO read(Long courseId);
    List<CourseDTO>readByAttribute(CourseAttribute courseAttribute);
    CourseDTO readReview(Long courseId);
    CourseDTO updateCourse(CourseDTO courseDTO);
    void deleteCourse(Long courseId);
    List<CourseDTO> readAll();
    List<CourseDTO> getCoursesByNickname(String nickname);
    List<MemberCourseDTO> getMemberCoursesByMemberId(Long memberId);
    List<CourseDTO> getCoursesByMemberId(Long memberId);

}

File: src/main/kotlin/edu/example/learner_kotlin/courseabout/course/service/CourseService.kt
package edu.example.learner_kotlin.courseabout.course.service

import edu.example.learner_kotlin.courseabout.course.dto.CourseCreateDTO
import edu.example.learner_kotlin.courseabout.course.dto.CourseDTO
import edu.example.learner_kotlin.courseabout.course.dto.CourseUpdateDTO
import edu.example.learner_kotlin.courseabout.course.dto.MemberCourseDTO
import edu.example.learner_kotlin.courseabout.course.entity.CourseAttribute
import org.springframework.stereotype.Service


interface CourseService {
    fun addCourse(dto: CourseCreateDTO): CourseCreateDTO
    fun read(courseId: Long): CourseDTO
    fun readByAttribute(courseAttribute: CourseAttribute): List<CourseDTO?>
    fun readReview(courseId: Long): CourseDTO
    fun updateCourse(dto: CourseUpdateDTO): CourseUpdateDTO
    fun deleteCourse(courseId: Long)
    fun readAll(): List<CourseDTO>
    fun getCoursesByNickname(nickname: String): List<CourseDTO>
    fun getMemberCoursesByMemberId(memberId: Long): List<MemberCourseDTO>
    fun getCoursesByMemberId(memberId: Long): List<CourseDTO>
}
